Firebase/Messaging/FIRMessaging.m

```diff
@@ -471,8 +471,9 @@ - (BOOL)isAutoInitEnabled {
   if (isAutoInitEnabledObject) {
     return [isAutoInitEnabledObject boolValue];
   }
-  // If none of above exists, we default assume FCM auto init is enabled.
-  return YES;
+
+  // If none of above exists, we default to the global switch that comes from FIRApp.
+  return self.isGlobalAutomaticDataCollectionEnabled;
```

Comment:
Before we make sure it's default true if no special action happened on the user side, so we make sure we are not breaking app's existing behavior on fetching instanceID, and other activities.

Is this global boolean value also serves the same purpose, that if developer doesn't do anything, it will be default ture?

Reply:
Yep, if nothing is set in the plist or at runtime, the default is true and we have unit tests to confirm that.